packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -132,23 +132,38 @@ export function createAlgoliaInsightsPlugin(
       return;
     }
 
-    const items = state.collections
-      .reduce<unknown[]>((acc, current) => {
-        return [...acc, ...current.items];
-      }, [])
-      .filter(isAlgoliaInsightsHit);
+    const nextItems = state.collections.reduce<
+      Record<string, AlgoliaInsightsHit[]>
+    >((acc, current) => {
+      const id = current.source.sourceId;
+      if (!acc[id]) {
+        acc[id] = [];
+      }
+      acc[id].push(...current.items.filter(isAlgoliaInsightsHit));
+      return acc;
+    }, {});
 
-    if (
-      !isEqual(
-        previousItems.current.map((x) => x.objectID),
-        items.map((x) => x.objectID)
-      )
-    ) {
-      previousItems.current = items;
+    const viewedItems: AlgoliaInsightsHit[] = [];
 
-      if (items.length > 0) {
-        sendViewedObjectIDs({ onItemsChange, items, insights, state });
+    Object.entries(nextItems).forEach(([id, items]) => {
+      if (
+        !isEqual(
+          (previousItems.current[id] || []).map((x) => x.objectID),
+          items.map((x) => x.objectID)
+        )
+      ) {
+        previousItems.current[id] = items;
+        viewedItems.push(...items);
       }
+    });
```

Comment:
One side effect of going through sources separately, is that no view event will be sent for a source if it hasn't changed, even if other sources have changed in response to a query. Events were still sent with the previous method.

The impact is that less view events will be sent for items in sources that don't change after a query update.

I think it's a non-issue, as it won't have a real impact, but I'd like to have your and other input on this.

Reply:
That's true, you can see that happening by for example using a space as the next query. I think that really we should be basing this comparison on `queryID` and thus don't need to compare so frequently, but I don't think that information is always available.

Reply:
Would something like this make sense, to send items from all sources whenever one of them has changed?

```ts
let sendEvents = false;
const nextItems = state.collections.reduce<
  Record<string, AlgoliaInsightsHit[]>
>((acc, current) => {
  const id = current.source.sourceId;
  if (!acc[id]) {
    acc[id] = [];
  }
  
  const filteredItems = current.items.filter(isAlgoliaInsightsHit);
  acc[id].push(...filteredItems);

  if (!sendEvents && !isEqual(
    (previousItems.current[id] || []).map((x) => x.objectID),
    filteredItems.map((x) => x.objectID)
  )) {
    sendEvents = true
  }

  return acc;
}, {});
```

Then later on you wouldn't need to recreate the `viewedItems` array and instead flatmap the array from `Object.values(nextItems)...` if `sendEvents` is true.

Wdyt?

Reply:
Technically that would always be true if there's more than one collection with the same source id right @dhayab?

Reply:
This would be, although I believe `sourceId` needs to be unique. But I don't think it is enforced or warned by the library.